server/models/recipe.js

```diff
@@ -20,24 +20,23 @@ module.exports = (sequelize, DataTypes) => {
     },
     {}
   );
-  Recipe.afterCreate(async recipe => {
-    const updatedRecipe = await recipe.update(
-      { slug: `${recipe.slug}-${recipe.id}` },
-      { returning: true }
-    );
-    Object.assign(recipe, updatedRecipe);
+
+  Recipe.beforeCreate(recipe => {
+    recipe.slug = `${slugifyTitle(recipe.title)}-${randomHex().substr(0, 8)}`;
```

Comment:
Why is this change

1) introduced
2) in this PR
3) in this location?

Reply:
1.  The change was introduced to avoid hitting the database twice (first to create a recipe, then to update the recipe's slug).

2. Unique slugs were first added to this codebase by this PR. 

3. I also reasoned that unique slug generation should be handled all in one place right before saving the recipe into the database

Reply:
Unique slugs were actually introduced in a different PR, as evidenced by the fact that they are already present in the develop branch. Here you are proposing a change to code that has already been merged. 

In essence be sure to pay attention to the commits you've just rebased your branch onto, not just accept or reject the changed from them. It should be clear, either from the PR or from code comments, why_modifications_ to existing code are being made, and preferably someone shouldn't have to ask :)

Reply:
Okay, got that. 
The PR description has been updated to reflect the reasons behind the modifications made.